streampipes-processors-transformation-jvm/src/main/java/org/apache/streampipes/processors/transformation/jvm/processor/booloperator/logical/BooleanOperatorProcessor.java

```diff
@@ -42,7 +44,14 @@ public DataProcessorDescription declareModel() {
                                 .create()
                                 .requiredProperty(EpRequirements.anyProperty())
                                 .build())
-                .requiredTextParameter(Labels.withId(BOOLEAN_PROCESSOR_INPUT_KEY))
+                .requiredSingleValueSelection(Labels.withId(BOOLEAN_OPERATOR_TYPE), Options.from(
+                        BooleanOperatorType.AND.operator(),
+                        BooleanOperatorType.OR.operator(),
+                        BooleanOperatorType.NOT.operator(),
+                        BooleanOperatorType.XOR.operator(),
+                        BooleanOperatorType.X_NOR.operator(),
+                        BooleanOperatorType.NOR.operator()))
```

Comment:
Great, that looks fine.

To let the user select a set of properties, see the suggestion in line 45.

Reply:
Thank you i've updated code. Please have a look again.

Reply:
This looks good
I will test it later briefly